components/getLocation.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import * as firebase from 'firebase';
+
+const MAPSAPI = 'https://maps.googleapis.com/maps/api/distancematrix/json?';
+const APIKEY = 'AIzaSyD2blk49vwxtYbL4h2TYrT_QlG_5UnwIKM';
+
+async function getData(url) {
+  const getUrl = await fetch(url);
+  const json = await getUrl.json();
+  return json; // parses response to JSON
+}
+
+async function GetAllDistances(latitude, longitude, uRad, locations) {
+  const validLocs = [];
+  for (const element of locations) {
+    const url = `${MAPSAPI}origins=${latitude},${longitude}&destinations=${
+      element.coordinates.latitude
+    },${element.coordinates.longitude}&${APIKEY}&mode=walking`;
+    let newData;
+    try {
+      newData = await getData(url);
```

Comment:
await in a loop throws a linting error, as having each loop iteration wait for the data is slow. Should refactor to use Promise.All(), so all the fetch requests start in parallel and get added to the loop when finished.